File: 2-medium/00056-merge-intervals/solution.cpp
class Solution {
    vector<vector<int>> res;
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {

        sort(intervals.begin(), intervals.end(), [](vector<int> &a, vector<int> &b){
            return a[0] < b[0];
        });

        int index = 0;
        while (index != intervals.size()) {
            int right = index + 1;
            auto &tmp = intervals[index];
            while (right < intervals.size() && intervals[right][0] <= tmp[1]){
                tmp[1] = std::max(tmp[1], intervals[right][1]);
                ++right;
            }
            res.push_back(tmp);
            index = right;
        }

        return res;
    }
};

File: 2-medium/00056-merge-intervals/solution.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:

        intervals.sort(key=lambda x: x[0])

        index = 0
        res = []

        while index != len(intervals):
            tmp_list = intervals[index][:]
            right = index + 1
            while right != len(intervals) and intervals[right][0] <= tmp_list[1]:
                tmp_list[1] = max(tmp_list[1], intervals[right][1])
                right += 1
            res.append(tmp_list)
            index = right

        return res        
